bundles/org.openhab.binding.caddx/src/main/java/org/openhab/binding/caddx/internal/handler/ThingHandlerPanel.java

```diff
@@ -0,0 +1,199 @@
+ */
+package org.openhab.binding.caddx.internal.handler;
+
+import java.util.HashMap;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.openhab.binding.caddx.internal.CaddxBindingConstants;
+import org.openhab.binding.caddx.internal.CaddxEvent;
+import org.openhab.binding.caddx.internal.CaddxMessage;
+import org.openhab.binding.caddx.internal.CaddxMessageType;
+import org.openhab.binding.caddx.internal.CaddxProperty;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This is a class for handling a Panel type Thing.
+ *
+ * @author Georgios Moutsos - Initial contribution
+ */
+@NonNullByDefault
+public class ThingHandlerPanel extends CaddxBaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(ThingHandlerPanel.class);
+
+    /**
+     * Constructor.
+     *
+     * @param thing
+     */
+    public ThingHandlerPanel(Thing thing) {
+        super(thing, CaddxThingType.PANEL);
+    }
+
+    @Override
+    public void updateChannel(ChannelUID channelUID, String data) {
+        if (channelUID.getId().equals(CaddxBindingConstants.PANEL_FIRMWARE_VERSION)
+                || channelUID.getId().startsWith("panel_log_message_")) {
+            updateState(channelUID, new StringType(data));
+        } else {
+            // All Panel channels are OnOffType
+            OnOffType onOffType;
+
+            onOffType = ("true".equals(data)) ? OnOffType.ON : OnOffType.OFF;
+            updateState(channelUID, onOffType);
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.trace("handleCommand(): Command Received - {} {}.", channelUID, command);
+
+        String cmd = null;
+        String data = null;
+        CaddxBridgeHandler bridgeHandler = getCaddxBridgeHandler();
+        if (bridgeHandler == null) {
+            return;
+        }
+
+        if (command instanceof RefreshType) {
+            if (CaddxBindingConstants.PANEL_FIRMWARE_VERSION.equals(channelUID.getId())) {
+                cmd = CaddxBindingConstants.PANEL_INTERFACE_CONFIGURATION_REQUEST;
+                data = "";
+            } else if (CaddxBindingConstants.PANEL_LOG_MESSAGE_N_0.equals(channelUID.getId())) {
+                cmd = CaddxBindingConstants.PANEL_SYSTEM_STATUS_REQUEST;
+                data = "";
+            } else {
+                return;
+            }
+        } else {
+            throw new IllegalArgumentException("Unknown command");
+        }
+
+        bridgeHandler.sendCommand(cmd, data);
```

Comment:
So `data` is only supposed to be `null` or `""` when it reaches this point?

Reply:
Actually only ""